// auth0/dist/esm/management/Client.mjs
// This file was auto-generated by Fern from our API Definition.
import { Actions } from "./api/resources/actions/client/Client.mjs";
import { Branding } from "./api/resources/branding/client/Client.mjs";
import { ClientGrants } from "./api/resources/clientGrants/client/Client.mjs";
import { Clients } from "./api/resources/clients/client/Client.mjs";
import { Connections } from "./api/resources/connections/client/Client.mjs";
import { CustomDomains } from "./api/resources/customDomains/client/Client.mjs";
import { DeviceCredentials } from "./api/resources/deviceCredentials/client/Client.mjs";
import { EmailTemplates } from "./api/resources/emailTemplates/client/Client.mjs";
import { EventStreams } from "./api/resources/eventStreams/client/Client.mjs";
import { Flows } from "./api/resources/flows/client/Client.mjs";
import { Forms } from "./api/resources/forms/client/Client.mjs";
import { UserGrants } from "./api/resources/userGrants/client/Client.mjs";
import { Hooks } from "./api/resources/hooks/client/Client.mjs";
import { Jobs } from "./api/resources/jobs/client/Client.mjs";
import { LogStreams } from "./api/resources/logStreams/client/Client.mjs";
import { Logs } from "./api/resources/logs/client/Client.mjs";
import { NetworkAcls } from "./api/resources/networkAcls/client/Client.mjs";
import { Organizations } from "./api/resources/organizations/client/Client.mjs";
import { Prompts } from "./api/resources/prompts/client/Client.mjs";
import { RefreshTokens } from "./api/resources/refreshTokens/client/Client.mjs";
import { ResourceServers } from "./api/resources/resourceServers/client/Client.mjs";
import { Roles } from "./api/resources/roles/client/Client.mjs";
import { Rules } from "./api/resources/rules/client/Client.mjs";
import { RulesConfigs } from "./api/resources/rulesConfigs/client/Client.mjs";
import { SelfServiceProfiles } from "./api/resources/selfServiceProfiles/client/Client.mjs";
import { Sessions } from "./api/resources/sessions/client/Client.mjs";
import { Stats } from "./api/resources/stats/client/Client.mjs";
import { SupplementalSignals } from "./api/resources/supplementalSignals/client/Client.mjs";
import { Tickets } from "./api/resources/tickets/client/Client.mjs";
import { TokenExchangeProfiles } from "./api/resources/tokenExchangeProfiles/client/Client.mjs";
import { UserAttributeProfiles } from "./api/resources/userAttributeProfiles/client/Client.mjs";
import { UserBlocks } from "./api/resources/userBlocks/client/Client.mjs";
import { Users } from "./api/resources/users/client/Client.mjs";
import { Anomaly } from "./api/resources/anomaly/client/Client.mjs";
import { AttackProtection } from "./api/resources/attackProtection/client/Client.mjs";
import { Emails } from "./api/resources/emails/client/Client.mjs";
import { Guardian } from "./api/resources/guardian/client/Client.mjs";
import { Keys } from "./api/resources/keys/client/Client.mjs";
import { RiskAssessments } from "./api/resources/riskAssessments/client/Client.mjs";
import { Tenants } from "./api/resources/tenants/client/Client.mjs";
import { VerifiableCredentials } from "./api/resources/verifiableCredentials/client/Client.mjs";
export class ManagementClient {
    constructor(_options) {
        this._options = _options;
    }
    get actions() {
        var _a;
        return ((_a = this._actions) !== null && _a !== void 0 ? _a : (this._actions = new Actions(this._options)));
    }
    get branding() {
        var _a;
        return ((_a = this._branding) !== null && _a !== void 0 ? _a : (this._branding = new Branding(this._options)));
    }
    get clientGrants() {
        var _a;
        return ((_a = this._clientGrants) !== null && _a !== void 0 ? _a : (this._clientGrants = new ClientGrants(this._options)));
    }
    get clients() {
        var _a;
        return ((_a = this._clients) !== null && _a !== void 0 ? _a : (this._clients = new Clients(this._options)));
    }
    get connections() {
        var _a;
        return ((_a = this._connections) !== null && _a !== void 0 ? _a : (this._connections = new Connections(this._options)));
    }
    get customDomains() {
        var _a;
        return ((_a = this._customDomains) !== null && _a !== void 0 ? _a : (this._customDomains = new CustomDomains(this._options)));
    }
    get deviceCredentials() {
        var _a;
        return ((_a = this._deviceCredentials) !== null && _a !== void 0 ? _a : (this._deviceCredentials = new DeviceCredentials(this._options)));
    }
    get emailTemplates() {
        var _a;
        return ((_a = this._emailTemplates) !== null && _a !== void 0 ? _a : (this._emailTemplates = new EmailTemplates(this._options)));
    }
    get eventStreams() {
        var _a;
        return ((_a = this._eventStreams) !== null && _a !== void 0 ? _a : (this._eventStreams = new EventStreams(this._options)));
    }
    get flows() {
        var _a;
        return ((_a = this._flows) !== null && _a !== void 0 ? _a : (this._flows = new Flows(this._options)));
    }
    get forms() {
        var _a;
        return ((_a = this._forms) !== null && _a !== void 0 ? _a : (this._forms = new Forms(this._options)));
    }
    get userGrants() {
        var _a;
        return ((_a = this._userGrants) !== null && _a !== void 0 ? _a : (this._userGrants = new UserGrants(this._options)));
    }
    get hooks() {
        var _a;
        return ((_a = this._hooks) !== null && _a !== void 0 ? _a : (this._hooks = new Hooks(this._options)));
    }
    get jobs() {
        var _a;
        return ((_a = this._jobs) !== null && _a !== void 0 ? _a : (this._jobs = new Jobs(this._options)));
    }
    get logStreams() {
        var _a;
        return ((_a = this._logStreams) !== null && _a !== void 0 ? _a : (this._logStreams = new LogStreams(this._options)));
    }
    get logs() {
        var _a;
        return ((_a = this._logs) !== null && _a !== void 0 ? _a : (this._logs = new Logs(this._options)));
    }
    get networkAcls() {
        var _a;
        return ((_a = this._networkAcls) !== null && _a !== void 0 ? _a : (this._networkAcls = new NetworkAcls(this._options)));
    }
    get organizations() {
        var _a;
        return ((_a = this._organizations) !== null && _a !== void 0 ? _a : (this._organizations = new Organizations(this._options)));
    }
    get prompts() {
        var _a;
        return ((_a = this._prompts) !== null && _a !== void 0 ? _a : (this._prompts = new Prompts(this._options)));
    }
    get refreshTokens() {
        var _a;
        return ((_a = this._refreshTokens) !== null && _a !== void 0 ? _a : (this._refreshTokens = new RefreshTokens(this._options)));
    }
    get resourceServers() {
        var _a;
        return ((_a = this._resourceServers) !== null && _a !== void 0 ? _a : (this._resourceServers = new ResourceServers(this._options)));
    }
    get roles() {
        var _a;
        return ((_a = this._roles) !== null && _a !== void 0 ? _a : (this._roles = new Roles(this._options)));
    }
    get rules() {
        var _a;
        return ((_a = this._rules) !== null && _a !== void 0 ? _a : (this._rules = new Rules(this._options)));
    }
    get rulesConfigs() {
        var _a;
        return ((_a = this._rulesConfigs) !== null && _a !== void 0 ? _a : (this._rulesConfigs = new RulesConfigs(this._options)));
    }
    get selfServiceProfiles() {
        var _a;
        return ((_a = this._selfServiceProfiles) !== null && _a !== void 0 ? _a : (this._selfServiceProfiles = new SelfServiceProfiles(this._options)));
    }
    get sessions() {
        var _a;
        return ((_a = this._sessions) !== null && _a !== void 0 ? _a : (this._sessions = new Sessions(this._options)));
    }
    get stats() {
        var _a;
        return ((_a = this._stats) !== null && _a !== void 0 ? _a : (this._stats = new Stats(this._options)));
    }
    get supplementalSignals() {
        var _a;
        return ((_a = this._supplementalSignals) !== null && _a !== void 0 ? _a : (this._supplementalSignals = new SupplementalSignals(this._options)));
    }
    get tickets() {
        var _a;
        return ((_a = this._tickets) !== null && _a !== void 0 ? _a : (this._tickets = new Tickets(this._options)));
    }
    get tokenExchangeProfiles() {
        var _a;
        return ((_a = this._tokenExchangeProfiles) !== null && _a !== void 0 ? _a : (this._tokenExchangeProfiles = new TokenExchangeProfiles(this._options)));
    }
    get userAttributeProfiles() {
        var _a;
        return ((_a = this._userAttributeProfiles) !== null && _a !== void 0 ? _a : (this._userAttributeProfiles = new UserAttributeProfiles(this._options)));
    }
    get userBlocks() {
        var _a;
        return ((_a = this._userBlocks) !== null && _a !== void 0 ? _a : (this._userBlocks = new UserBlocks(this._options)));
    }
    get users() {
        var _a;
        return ((_a = this._users) !== null && _a !== void 0 ? _a : (this._users = new Users(this._options)));
    }
    get anomaly() {
        var _a;
        return ((_a = this._anomaly) !== null && _a !== void 0 ? _a : (this._anomaly = new Anomaly(this._options)));
    }
    get attackProtection() {
        var _a;
        return ((_a = this._attackProtection) !== null && _a !== void 0 ? _a : (this._attackProtection = new AttackProtection(this._options)));
    }
    get emails() {
        var _a;
        return ((_a = this._emails) !== null && _a !== void 0 ? _a : (this._emails = new Emails(this._options)));
    }
    get guardian() {
        var _a;
        return ((_a = this._guardian) !== null && _a !== void 0 ? _a : (this._guardian = new Guardian(this._options)));
    }
    get keys() {
        var _a;
        return ((_a = this._keys) !== null && _a !== void 0 ? _a : (this._keys = new Keys(this._options)));
    }
    get riskAssessments() {
        var _a;
        return ((_a = this._riskAssessments) !== null && _a !== void 0 ? _a : (this._riskAssessments = new RiskAssessments(this._options)));
    }
    get tenants() {
        var _a;
        return ((_a = this._tenants) !== null && _a !== void 0 ? _a : (this._tenants = new Tenants(this._options)));
    }
    get verifiableCredentials() {
        var _a;
        return ((_a = this._verifiableCredentials) !== null && _a !== void 0 ? _a : (this._verifiableCredentials = new VerifiableCredentials(this._options)));
    }
}
