// auth0/dist/esm/management/api/resources/attackProtection/client/Client.mjs
// This file was auto-generated by Fern from our API Definition.
import { BotDetection } from "../resources/botDetection/client/Client.mjs";
import { BreachedPasswordDetection } from "../resources/breachedPasswordDetection/client/Client.mjs";
import { BruteForceProtection } from "../resources/bruteForceProtection/client/Client.mjs";
import { Captcha } from "../resources/captcha/client/Client.mjs";
import { SuspiciousIpThrottling } from "../resources/suspiciousIpThrottling/client/Client.mjs";
export class AttackProtection {
    constructor(_options) {
        this._options = _options;
    }
    get botDetection() {
        var _a;
        return ((_a = this._botDetection) !== null && _a !== void 0 ? _a : (this._botDetection = new BotDetection(this._options)));
    }
    get breachedPasswordDetection() {
        var _a;
        return ((_a = this._breachedPasswordDetection) !== null && _a !== void 0 ? _a : (this._breachedPasswordDetection = new BreachedPasswordDetection(this._options)));
    }
    get bruteForceProtection() {
        var _a;
        return ((_a = this._bruteForceProtection) !== null && _a !== void 0 ? _a : (this._bruteForceProtection = new BruteForceProtection(this._options)));
    }
    get captcha() {
        var _a;
        return ((_a = this._captcha) !== null && _a !== void 0 ? _a : (this._captcha = new Captcha(this._options)));
    }
    get suspiciousIpThrottling() {
        var _a;
        return ((_a = this._suspiciousIpThrottling) !== null && _a !== void 0 ? _a : (this._suspiciousIpThrottling = new SuspiciousIpThrottling(this._options)));
    }
}
