// auth0/dist/esm/management/api/types/types.mjs
// This file was auto-generated by Fern from our API Definition.
export const OauthScope = {
    /**
     * Read Actions */
    ReadActions: "read:actions",
    /**
     * Create Actions */
    CreateActions: "create:actions",
    /**
     * Delete Actions */
    DeleteActions: "delete:actions",
    /**
     * Update Actions */
    UpdateActions: "update:actions",
    /**
     * Read Anomaly Blocks */
    ReadAnomalyBlocks: "read:anomaly_blocks",
    /**
     * Delete Anomaly Blocks */
    DeleteAnomalyBlocks: "delete:anomaly_blocks",
    /**
     * Read Shields */
    ReadShields: "read:shields",
    /**
     * Create Shields */
    CreateShields: "create:shields",
    /**
     * Update Shields */
    UpdateShields: "update:shields",
    /**
     * Read Attack Protection */
    ReadAttackProtection: "read:attack_protection",
    /**
     * Update Attack Protection */
    UpdateAttackProtection: "update:attack_protection",
    /**
     * Read Branding */
    ReadBranding: "read:branding",
    /**
     * Update Branding */
    UpdateBranding: "update:branding",
    /**
     * Read Phone Providers */
    ReadPhoneProviders: "read:phone_providers",
    /**
     * Create Phone Providers */
    CreatePhoneProviders: "create:phone_providers",
    /**
     * Update Phone Providers */
    UpdatePhoneProviders: "update:phone_providers",
    /**
     * Delete Phone Providers */
    DeletePhoneProviders: "delete:phone_providers",
    /**
     * Read Phone Templates */
    ReadPhoneTemplates: "read:phone_templates",
    /**
     * Create Phone Templates */
    CreatePhoneTemplates: "create:phone_templates",
    /**
     * Update Phone Templates */
    UpdatePhoneTemplates: "update:phone_templates",
    /**
     * Delete Phone Templates */
    DeletePhoneTemplates: "delete:phone_templates",
    /**
     * Delete Branding */
    DeleteBranding: "delete:branding",
    /**
     * Read Client Grants */
    ReadClientGrants: "read:client_grants",
    /**
     * Create Client Grants */
    CreateClientGrants: "create:client_grants",
    /**
     * Update Client Grants */
    UpdateClientGrants: "update:client_grants",
    /**
     * Delete Client Grants */
    DeleteClientGrants: "delete:client_grants",
    /**
     * Read Organization Client Grants */
    ReadOrganizationClientGrants: "read:organization_client_grants",
    /**
     * Read Clients */
    ReadClients: "read:clients",
    /**
     * Read Client Keys */
    ReadClientKeys: "read:client_keys",
    /**
     * Read Client Credentials */
    ReadClientCredentials: "read:client_credentials",
    /**
     * Read Client Summary */
    ReadClientSummary: "read:client_summary",
    /**
     * Create Clients */
    CreateClients: "create:clients",
    /**
     * Create Client Credentials */
    CreateClientCredentials: "create:client_credentials",
    /**
     * Update Client Credentials */
    UpdateClientCredentials: "update:client_credentials",
    /**
     * Delete Client Credentials */
    DeleteClientCredentials: "delete:client_credentials",
    /**
     * Update Clients */
    UpdateClients: "update:clients",
    /**
     * Update Client Keys */
    UpdateClientKeys: "update:client_keys",
    /**
     * Update Client Token Vault Privileged Access */
    UpdateClientTokenVaultPrivilegedAccess: "update:client_token_vault_privileged_access",
    /**
     * Delete Clients */
    DeleteClients: "delete:clients",
    /**
     * Read Connections */
    ReadConnections: "read:connections",
    /**
     * Read Connection Profiles */
    ReadConnectionProfiles: "read:connection_profiles",
    /**
     * Create Connection Profiles */
    CreateConnectionProfiles: "create:connection_profiles",
    /**
     * Update Connection Profiles */
    UpdateConnectionProfiles: "update:connection_profiles",
    /**
     * Delete Connection Profiles */
    DeleteConnectionProfiles: "delete:connection_profiles",
    /**
     * Create Connections */
    CreateConnections: "create:connections",
    /**
     * Update Connections */
    UpdateConnections: "update:connections",
    /**
     * Delete Connections */
    DeleteConnections: "delete:connections",
    /**
     * Read Directory Provisionings */
    ReadDirectoryProvisionings: "read:directory_provisionings",
    /**
     * Create Directory Provisionings */
    CreateDirectoryProvisionings: "create:directory_provisionings",
    /**
     * Update Directory Provisionings */
    UpdateDirectoryProvisionings: "update:directory_provisionings",
    /**
     * Delete Directory Provisionings */
    DeleteDirectoryProvisionings: "delete:directory_provisionings",
    /**
     * Read Users */
    ReadUsers: "read:users",
    /**
     * Read Connections Keys */
    ReadConnectionsKeys: "read:connections_keys",
    /**
     * Create Connections Keys */
    CreateConnectionsKeys: "create:connections_keys",
    /**
     * Update Connections Keys */
    UpdateConnectionsKeys: "update:connections_keys",
    /**
     * Read Scim Config */
    ReadScimConfig: "read:scim_config",
    /**
     * Create Scim Config */
    CreateScimConfig: "create:scim_config",
    /**
     * Update Scim Config */
    UpdateScimConfig: "update:scim_config",
    /**
     * Delete Scim Config */
    DeleteScimConfig: "delete:scim_config",
    /**
     * Read Scim Token */
    ReadScimToken: "read:scim_token",
    /**
     * Create Scim Token */
    CreateScimToken: "create:scim_token",
    /**
     * Delete Scim Token */
    DeleteScimToken: "delete:scim_token",
    /**
     * Delete Users */
    DeleteUsers: "delete:users",
    /**
     * Read Custom Domains */
    ReadCustomDomains: "read:custom_domains",
    /**
     * Create Custom Domains */
    CreateCustomDomains: "create:custom_domains",
    /**
     * Update Custom Domains */
    UpdateCustomDomains: "update:custom_domains",
    /**
     * Delete Custom Domains */
    DeleteCustomDomains: "delete:custom_domains",
    /**
     * Read Device Credentials */
    ReadDeviceCredentials: "read:device_credentials",
    /**
     * Create Current User Device Credentials */
    CreateCurrentUserDeviceCredentials: "create:current_user_device_credentials",
    /**
     * Delete Device Credentials */
    DeleteDeviceCredentials: "delete:device_credentials",
    /**
     * Delete Current User Device Credentials */
    DeleteCurrentUserDeviceCredentials: "delete:current_user_device_credentials",
    /**
     * Update Device Codes */
    UpdateDeviceCodes: "update:device_codes",
    /**
     * Read Device Codes */
    ReadDeviceCodes: "read:device_codes",
    /**
     * Create Test Email Dispatch */
    CreateTestEmailDispatch: "create:test_email_dispatch",
    /**
     * Create Email Templates */
    CreateEmailTemplates: "create:email_templates",
    /**
     * Read Email Templates */
    ReadEmailTemplates: "read:email_templates",
    /**
     * Update Email Templates */
    UpdateEmailTemplates: "update:email_templates",
    /**
     * Read Email Provider */
    ReadEmailProvider: "read:email_provider",
    /**
     * Create Email Provider */
    CreateEmailProvider: "create:email_provider",
    /**
     * Update Email Provider */
    UpdateEmailProvider: "update:email_provider",
    /**
     * Delete Email Provider */
    DeleteEmailProvider: "delete:email_provider",
    /**
     * Read Entitlements */
    ReadEntitlements: "read:entitlements",
    /**
     * Read Event Streams */
    ReadEventStreams: "read:event_streams",
    /**
     * Create Event Streams */
    CreateEventStreams: "create:event_streams",
    /**
     * Update Event Streams */
    UpdateEventStreams: "update:event_streams",
    /**
     * Delete Event Streams */
    DeleteEventStreams: "delete:event_streams",
    /**
     * Read Event Deliveries */
    ReadEventDeliveries: "read:event_deliveries",
    /**
     * Update Event Deliveries */
    UpdateEventDeliveries: "update:event_deliveries",
    /**
     * Read Extensions */
    ReadExtensions: "read:extensions",
    /**
     * Read Flows */
    ReadFlows: "read:flows",
    /**
     * Create Flows */
    CreateFlows: "create:flows",
    /**
     * Read Flows Vault Connections */
    ReadFlowsVaultConnections: "read:flows_vault_connections",
    /**
     * Create Flows Vault Connections */
    CreateFlowsVaultConnections: "create:flows_vault_connections",
    /**
     * Update Flows Vault Connections */
    UpdateFlowsVaultConnections: "update:flows_vault_connections",
    /**
     * Delete Flows Vault Connections */
    DeleteFlowsVaultConnections: "delete:flows_vault_connections",
    /**
     * Read Flows Executions */
    ReadFlowsExecutions: "read:flows_executions",
    /**
     * Delete Flows Executions */
    DeleteFlowsExecutions: "delete:flows_executions",
    /**
     * Update Flows */
    UpdateFlows: "update:flows",
    /**
     * Delete Flows */
    DeleteFlows: "delete:flows",
    /**
     * Read Forms */
    ReadForms: "read:forms",
    /**
     * Create Forms */
    CreateForms: "create:forms",
    /**
     * Update Forms */
    UpdateForms: "update:forms",
    /**
     * Delete Forms */
    DeleteForms: "delete:forms",
    /**
     * Read Grants */
    ReadGrants: "read:grants",
    /**
     * Delete Grants */
    DeleteGrants: "delete:grants",
    /**
     * Read Groups */
    ReadGroups: "read:groups",
    /**
     * Read Group Members */
    ReadGroupMembers: "read:group_members",
    /**
     * Create Guardian Enrollment Tickets */
    CreateGuardianEnrollmentTickets: "create:guardian_enrollment_tickets",
    /**
     * Read Guardian Enrollments */
    ReadGuardianEnrollments: "read:guardian_enrollments",
    /**
     * Delete Guardian Enrollments */
    DeleteGuardianEnrollments: "delete:guardian_enrollments",
    /**
     * Read Guardian Factors */
    ReadGuardianFactors: "read:guardian_factors",
    /**
     * Update Guardian Factors */
    UpdateGuardianFactors: "update:guardian_factors",
    /**
     * Read Mfa Policies */
    ReadMfaPolicies: "read:mfa_policies",
    /**
     * Update Mfa Policies */
    UpdateMfaPolicies: "update:mfa_policies",
    /**
     * Read Hooks */
    ReadHooks: "read:hooks",
    /**
     * Create Hooks */
    CreateHooks: "create:hooks",
    /**
     * Update Hooks */
    UpdateHooks: "update:hooks",
    /**
     * Delete Hooks */
    DeleteHooks: "delete:hooks",
    /**
     * Read Insights */
    ReadInsights: "read:insights",
    /**
     * Read Stats */
    ReadStats: "read:stats",
    /**
     * Read Integrations */
    ReadIntegrations: "read:integrations",
    /**
     * Create Integrations */
    CreateIntegrations: "create:integrations",
    /**
     * Update Integrations */
    UpdateIntegrations: "update:integrations",
    /**
     * Delete Integrations */
    DeleteIntegrations: "delete:integrations",
    /**
     * Create Users */
    CreateUsers: "create:users",
    /**
     * Update Users */
    UpdateUsers: "update:users",
    /**
     * Read Custom Signing Keys */
    ReadCustomSigningKeys: "read:custom_signing_keys",
    /**
     * Create Custom Signing Keys */
    CreateCustomSigningKeys: "create:custom_signing_keys",
    /**
     * Update Custom Signing Keys */
    UpdateCustomSigningKeys: "update:custom_signing_keys",
    /**
     * Delete Custom Signing Keys */
    DeleteCustomSigningKeys: "delete:custom_signing_keys",
    /**
     * Read Encryption Keys */
    ReadEncryptionKeys: "read:encryption_keys",
    /**
     * Create Encryption Keys */
    CreateEncryptionKeys: "create:encryption_keys",
    /**
     * Update Encryption Keys */
    UpdateEncryptionKeys: "update:encryption_keys",
    /**
     * Delete Encryption Keys */
    DeleteEncryptionKeys: "delete:encryption_keys",
    /**
     * Read Signing Keys */
    ReadSigningKeys: "read:signing_keys",
    /**
     * Create Signing Keys */
    CreateSigningKeys: "create:signing_keys",
    /**
     * Update Signing Keys */
    UpdateSigningKeys: "update:signing_keys",
    /**
     * Read Log Streams */
    ReadLogStreams: "read:log_streams",
    /**
     * Create Log Streams */
    CreateLogStreams: "create:log_streams",
    /**
     * Update Log Streams */
    UpdateLogStreams: "update:log_streams",
    /**
     * Delete Log Streams */
    DeleteLogStreams: "delete:log_streams",
    /**
     * Read Logs */
    ReadLogs: "read:logs",
    /**
     * Read Logs Users */
    ReadLogsUsers: "read:logs_users",
    /**
     * Read Tenant Settings */
    ReadTenantSettings: "read:tenant_settings",
    /**
     * Update Tenant Settings */
    UpdateTenantSettings: "update:tenant_settings",
    /**
     * Read Network Acls */
    ReadNetworkAcls: "read:network_acls",
    /**
     * Create Network Acls */
    CreateNetworkAcls: "create:network_acls",
    /**
     * Update Network Acls */
    UpdateNetworkAcls: "update:network_acls",
    /**
     * Delete Network Acls */
    DeleteNetworkAcls: "delete:network_acls",
    /**
     * Read Organizations */
    ReadOrganizations: "read:organizations",
    /**
     * Read Organizations Summary */
    ReadOrganizationsSummary: "read:organizations_summary",
    /**
     * Create Organizations */
    CreateOrganizations: "create:organizations",
    /**
     * Create Organization Connections */
    CreateOrganizationConnections: "create:organization_connections",
    /**
     * Update Organizations */
    UpdateOrganizations: "update:organizations",
    /**
     * Delete Organizations */
    DeleteOrganizations: "delete:organizations",
    /**
     * Create Organization Client Grants */
    CreateOrganizationClientGrants: "create:organization_client_grants",
    /**
     * Delete Organization Client Grants */
    DeleteOrganizationClientGrants: "delete:organization_client_grants",
    /**
     * Read Organization Connections */
    ReadOrganizationConnections: "read:organization_connections",
    /**
     * Update Organization Connections */
    UpdateOrganizationConnections: "update:organization_connections",
    /**
     * Delete Organization Connections */
    DeleteOrganizationConnections: "delete:organization_connections",
    /**
     * Read Organization Discovery Domains */
    ReadOrganizationDiscoveryDomains: "read:organization_discovery_domains",
    /**
     * Create Organization Discovery Domains */
    CreateOrganizationDiscoveryDomains: "create:organization_discovery_domains",
    /**
     * Update Organization Discovery Domains */
    UpdateOrganizationDiscoveryDomains: "update:organization_discovery_domains",
    /**
     * Delete Organization Discovery Domains */
    DeleteOrganizationDiscoveryDomains: "delete:organization_discovery_domains",
    /**
     * Read Organization Invitations */
    ReadOrganizationInvitations: "read:organization_invitations",
    /**
     * Create Organization Invitations */
    CreateOrganizationInvitations: "create:organization_invitations",
    /**
     * Delete Organization Invitations */
    DeleteOrganizationInvitations: "delete:organization_invitations",
    /**
     * Read Organization Members */
    ReadOrganizationMembers: "read:organization_members",
    /**
     * Create Organization Members */
    CreateOrganizationMembers: "create:organization_members",
    /**
     * Delete Organization Members */
    DeleteOrganizationMembers: "delete:organization_members",
    /**
     * Read Organization Member Roles */
    ReadOrganizationMemberRoles: "read:organization_member_roles",
    /**
     * Create Organization Member Roles */
    CreateOrganizationMemberRoles: "create:organization_member_roles",
    /**
     * Delete Organization Member Roles */
    DeleteOrganizationMemberRoles: "delete:organization_member_roles",
    /**
     * Read Prompts */
    ReadPrompts: "read:prompts",
    /**
     * Update Prompts */
    UpdatePrompts: "update:prompts",
    /**
     * Read Resource Servers */
    ReadResourceServers: "read:resource_servers",
    /**
     * Update Resource Servers */
    UpdateResourceServers: "update:resource_servers",
    /**
     * Read Refresh Tokens */
    ReadRefreshTokens: "read:refresh_tokens",
    /**
     * Delete Refresh Tokens */
    DeleteRefreshTokens: "delete:refresh_tokens",
    /**
     * Create Resource Servers */
    CreateResourceServers: "create:resource_servers",
    /**
     * Delete Resource Servers */
    DeleteResourceServers: "delete:resource_servers",
    /**
     * Read Roles */
    ReadRoles: "read:roles",
    /**
     * Create Roles */
    CreateRoles: "create:roles",
    /**
     * Update Roles */
    UpdateRoles: "update:roles",
    /**
     * Delete Roles */
    DeleteRoles: "delete:roles",
    /**
     * Read Role Members */
    ReadRoleMembers: "read:role_members",
    /**
     * Create Role Members */
    CreateRoleMembers: "create:role_members",
    /**
     * Read Rules */
    ReadRules: "read:rules",
    /**
     * Create Rules */
    CreateRules: "create:rules",
    /**
     * Update Rules */
    UpdateRules: "update:rules",
    /**
     * Read Rules Configs */
    ReadRulesConfigs: "read:rules_configs",
    /**
     * Update Rules Configs */
    UpdateRulesConfigs: "update:rules_configs",
    /**
     * Delete Rules Configs */
    DeleteRulesConfigs: "delete:rules_configs",
    /**
     * Delete Rules */
    DeleteRules: "delete:rules",
    /**
     * Read Security Metrics */
    ReadSecurityMetrics: "read:security_metrics",
    /**
     * Read Self Service Profiles */
    ReadSelfServiceProfiles: "read:self_service_profiles",
    /**
     * Create Self Service Profiles */
    CreateSelfServiceProfiles: "create:self_service_profiles",
    /**
     * Update Self Service Profiles */
    UpdateSelfServiceProfiles: "update:self_service_profiles",
    /**
     * Delete Self Service Profiles */
    DeleteSelfServiceProfiles: "delete:self_service_profiles",
    /**
     * Read Self Service Profile Custom Texts */
    ReadSelfServiceProfileCustomTexts: "read:self_service_profile_custom_texts",
    /**
     * Update Self Service Profile Custom Texts */
    UpdateSelfServiceProfileCustomTexts: "update:self_service_profile_custom_texts",
    /**
     * Create Sso Access Tickets */
    CreateSsoAccessTickets: "create:sso_access_tickets",
    /**
     * Delete Sso Access Tickets */
    DeleteSsoAccessTickets: "delete:sso_access_tickets",
    /**
     * Read Sessions */
    ReadSessions: "read:sessions",
    /**
     * Update Sessions */
    UpdateSessions: "update:sessions",
    /**
     * Delete Sessions */
    DeleteSessions: "delete:sessions",
    /**
     * Delete Tenants */
    DeleteTenants: "delete:tenants",
    /**
     * Run Checks */
    RunChecks: "run:checks",
    /**
     * Read Checks */
    ReadChecks: "read:checks",
    /**
     * Read Tenant Feature Flags */
    ReadTenantFeatureFlags: "read:tenant_feature_flags",
    /**
     * Read Tenant Invitations */
    ReadTenantInvitations: "read:tenant_invitations",
    /**
     * Create Tenant Invitations */
    CreateTenantInvitations: "create:tenant_invitations",
    /**
     * Update Tenant Invitations */
    UpdateTenantInvitations: "update:tenant_invitations",
    /**
     * Delete Tenant Invitations */
    DeleteTenantInvitations: "delete:tenant_invitations",
    /**
     * Read Tenant Members */
    ReadTenantMembers: "read:tenant_members",
    /**
     * Update Tenant Members */
    UpdateTenantMembers: "update:tenant_members",
    /**
     * Delete Tenant Members */
    DeleteTenantMembers: "delete:tenant_members",
    /**
     * Read Owners */
    ReadOwners: "read:owners",
    /**
     * Delete Owners */
    DeleteOwners: "delete:owners",
    /**
     * Create User Tickets */
    CreateUserTickets: "create:user_tickets",
    /**
     * Read Token Exchange Profiles */
    ReadTokenExchangeProfiles: "read:token_exchange_profiles",
    /**
     * Create Token Exchange Profiles */
    CreateTokenExchangeProfiles: "create:token_exchange_profiles",
    /**
     * Update Token Exchange Profiles */
    UpdateTokenExchangeProfiles: "update:token_exchange_profiles",
    /**
     * Delete Token Exchange Profiles */
    DeleteTokenExchangeProfiles: "delete:token_exchange_profiles",
    /**
     * Read Entity Counts */
    ReadEntityCounts: "read:entity_counts",
    /**
     * Read User Attribute Profiles */
    ReadUserAttributeProfiles: "read:user_attribute_profiles",
    /**
     * Create User Attribute Profiles */
    CreateUserAttributeProfiles: "create:user_attribute_profiles",
    /**
     * Update User Attribute Profiles */
    UpdateUserAttributeProfiles: "update:user_attribute_profiles",
    /**
     * Delete User Attribute Profiles */
    DeleteUserAttributeProfiles: "delete:user_attribute_profiles",
    /**
     * Read User Idp Tokens */
    ReadUserIdpTokens: "read:user_idp_tokens",
    /**
     * Read Current User */
    ReadCurrentUser: "read:current_user",
    /**
     * Update Users App Metadata */
    UpdateUsersAppMetadata: "update:users_app_metadata",
    /**
     * Update Current User Metadata */
    UpdateCurrentUserMetadata: "update:current_user_metadata",
    /**
     * Delete Current User */
    DeleteCurrentUser: "delete:current_user",
    /**
     * Read User Application Passwords */
    ReadUserApplicationPasswords: "read:user_application_passwords",
    /**
     * Create User Application Passwords */
    CreateUserApplicationPasswords: "create:user_application_passwords",
    /**
     * Delete User Application Passwords */
    DeleteUserApplicationPasswords: "delete:user_application_passwords",
    /**
     * Read Authentication Methods */
    ReadAuthenticationMethods: "read:authentication_methods",
    /**
     * Update Authentication Methods */
    UpdateAuthenticationMethods: "update:authentication_methods",
    /**
     * Create Authentication Methods */
    CreateAuthenticationMethods: "create:authentication_methods",
    /**
     * Delete Authentication Methods */
    DeleteAuthenticationMethods: "delete:authentication_methods",
    /**
     * Read Federated Connections Tokens */
    ReadFederatedConnectionsTokens: "read:federated_connections_tokens",
    /**
     * Delete Federated Connections Tokens */
    DeleteFederatedConnectionsTokens: "delete:federated_connections_tokens",
    /**
     * Update Current User Identities */
    UpdateCurrentUserIdentities: "update:current_user_identities",
    /**
     * Delete Role Members */
    DeleteRoleMembers: "delete:role_members",
    /**
     * Read Vdcs Templates */
    ReadVdcsTemplates: "read:vdcs_templates",
    /**
     * Create Vdcs Templates */
    CreateVdcsTemplates: "create:vdcs_templates",
    /**
     * Update Vdcs Templates */
    UpdateVdcsTemplates: "update:vdcs_templates",
    /**
     * Delete Vdcs Templates */
    DeleteVdcsTemplates: "delete:vdcs_templates",
};
/** How the action is being referred to: `action_id` or `action_name`. */
export const ActionBindingRefTypeEnum = {
    BindingId: "binding_id",
    ActionId: "action_id",
    ActionName: "action_name",
};
/** In order to execute an Action, it must be bound to a trigger using a binding. `trigger-bound` means that bindings are managed by the tenant. `entity-bound` means that the bindings are automatically managed by Auth0 and other internal resouces will control those bindings. Tenants cannot manage `entity-bound` bindings. */
export const ActionBindingTypeEnum = {
    TriggerBound: "trigger-bound",
    EntityBound: "entity-bound",
};
/** The build status of this action. */
export const ActionBuildStatusEnum = {
    Pending: "pending",
    Building: "building",
    Packaged: "packaged",
    Built: "built",
    Retrying: "retrying",
    Failed: "failed",
};
/** The overall status of an execution. */
export const ActionExecutionStatusEnum = {
    Unspecified: "unspecified",
    Pending: "pending",
    Final: "final",
    Partial: "partial",
    Canceled: "canceled",
    Suspended: "suspended",
};
/** The build status of this specific version. */
export const ActionVersionBuildStatusEnum = {
    Pending: "pending",
    Building: "building",
    Packaged: "packaged",
    Built: "built",
    Retrying: "retrying",
    Failed: "failed",
};
/** Type of match to apply */
export const AculMatchTypeEnum = {
    IncludesAny: "includes_any",
    ExcludesAny: "excludes_any",
};
/** Rendering mode to filter by */
export const AculRenderingModeEnum = {
    Advanced: "advanced",
    Standard: "standard",
};
export const AsyncApprovalNotificationsChannelsEnum = {
    GuardianPush: "guardian-push",
    Email: "email",
};
/** The id of the active provider for the CAPTCHA. */
export const AttackProtectionCaptchaProviderId = {
    Arkose: "arkose",
    AuthChallenge: "auth_challenge",
    FriendlyCaptcha: "friendly_captcha",
    Hcaptcha: "hcaptcha",
    RecaptchaV2: "recaptcha_v2",
    RecaptchaEnterprise: "recaptcha_enterprise",
    SimpleCaptcha: "simple_captcha",
};
export const AuthenticationMethodTypeEnum = {
    RecoveryCode: "recovery-code",
    Totp: "totp",
    Push: "push",
    Phone: "phone",
    Email: "email",
    EmailVerification: "email-verification",
    WebauthnRoaming: "webauthn-roaming",
    WebauthnPlatform: "webauthn-platform",
    Guardian: "guardian",
    Passkey: "passkey",
    Password: "password",
};
export const AuthenticationTypeEnum = {
    Phone: "phone",
    Email: "email",
    Totp: "totp",
};
/** The policy that defines how often to show CAPTCHA */
export const BotDetectionChallengePolicyPasswordFlowEnum = {
    Never: "never",
    WhenRisky: "when_risky",
    Always: "always",
};
/** The policy that defines how often to show CAPTCHA */
export const BotDetectionChallengePolicyPasswordResetFlowEnum = {
    Never: "never",
    WhenRisky: "when_risky",
    Always: "always",
};
/** The policy that defines how often to show CAPTCHA */
export const BotDetectionChallengePolicyPasswordlessFlowEnum = {
    Never: "never",
    WhenRisky: "when_risky",
    Always: "always",
};
/** The level of bot detection sensitivity */
export const BotDetectionLevelEnum = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
/** Buttons style */
export const BrandingThemeBordersButtonsStyleEnum = {
    Pill: "pill",
    Rounded: "rounded",
    Sharp: "sharp",
};
/** Inputs style */
export const BrandingThemeBordersInputsStyleEnum = {
    Pill: "pill",
    Rounded: "rounded",
    Sharp: "sharp",
};
/** Captcha Widget Theme */
export const BrandingThemeColorsCaptchaWidgetThemeEnum = {
    Auto: "auto",
    Dark: "dark",
    Light: "light",
};
/** Links style */
export const BrandingThemeFontLinksStyleEnum = {
    Normal: "normal",
    Underlined: "underlined",
};
/** Page Layout */
export const BrandingThemePageBackgroundPageLayoutEnum = {
    Center: "center",
    Left: "left",
    Right: "right",
};
/** Header text alignment */
export const BrandingThemeWidgetHeaderTextAlignmentEnum = {
    Center: "center",
    Left: "left",
    Right: "right",
};
/** Logo position */
export const BrandingThemeWidgetLogoPositionEnum = {
    Center: "center",
    Left: "left",
    None: "none",
    Right: "right",
};
/** Social buttons layout */
export const BrandingThemeWidgetSocialButtonsLayoutEnum = {
    Bottom: "bottom",
    Top: "top",
};
export const BreachedPasswordDetectionAdminNotificationFrequencyEnum = {
    Immediately: "immediately",
    Daily: "daily",
    Weekly: "weekly",
    Monthly: "monthly",
};
/**
 * The subscription level for breached password detection methods. Use "enhanced" to enable Credential Guard.
 *         Possible values: <code>standard</code>, <code>enhanced</code>.
 */
export const BreachedPasswordDetectionMethodEnum = {
    Standard: "standard",
    Enhanced: "enhanced",
};
export const BreachedPasswordDetectionPreChangePasswordShieldsEnum = {
    Block: "block",
    AdminNotification: "admin_notification",
};
export const BreachedPasswordDetectionPreUserRegistrationShieldsEnum = {
    Block: "block",
    AdminNotification: "admin_notification",
};
export const BreachedPasswordDetectionShieldsEnum = {
    Block: "block",
    UserNotification: "user_notification",
    AdminNotification: "admin_notification",
};
/** The type of application this client represents */
export const ClientAppTypeEnum = {
    Native: "native",
    Spa: "spa",
    RegularWeb: "regular_web",
    NonInteractive: "non_interactive",
    ResourceServer: "resource_server",
    ExpressConfiguration: "express_configuration",
    Rms: "rms",
    Box: "box",
    Cloudbees: "cloudbees",
    Concur: "concur",
    Dropbox: "dropbox",
    Mscrm: "mscrm",
    Echosign: "echosign",
    Egnyte: "egnyte",
    Newrelic: "newrelic",
    Office365: "office365",
    Salesforce: "salesforce",
    Sentry: "sentry",
    Sharepoint: "sharepoint",
    Slack: "slack",
    Springcm: "springcm",
    Zendesk: "zendesk",
    Zoom: "zoom",
    SsoIntegration: "sso_integration",
    Oag: "oag",
};
/** Defines the compliance level for this client, which may restrict it's capabilities */
export const ClientComplianceLevelEnum = {
    None: "none",
    Fapi1AdvPkjPar: "fapi1_adv_pkj_par",
    Fapi1AdvMtlsPar: "fapi1_adv_mtls_par",
    Fapi2SpPkjMtls: "fapi2_sp_pkj_mtls",
    Fapi2SpMtlsMtls: "fapi2_sp_mtls_mtls",
};
/** Algorithm which will be used with the credential. Supported algorithms: RS256,RS384,PS256 */
export const ClientCredentialAlgorithmEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Ps256: "PS256",
};
/** The type of credential. */
export const ClientCredentialTypeEnum = {
    PublicKey: "public_key",
    CertSubjectDn: "cert_subject_dn",
    X509Cert: "x509_cert",
};
/** Controls how organizations may be used with this grant */
export const ClientGrantOrganizationNullableUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** Defines whether organizations can be used with client credentials exchanges for this grant. */
export const ClientGrantOrganizationUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** The type of application access the client grant allows. Use of this field is subject to the applicable Free Trial terms in Okta’s <a href="https://www.okta.com/legal/"> Master Subscription Agreement.</a> */
export const ClientGrantSubjectTypeEnum = {
    Client: "client",
    User: "user",
};
/** The `selected_initiators` property contains the list of initiators to be enabled for the given application. */
export const ClientOidcBackchannelLogoutInitiatorsEnum = {
    RpLogout: "rp-logout",
    IdpLogout: "idp-logout",
    PasswordChanged: "password-changed",
    SessionExpired: "session-expired",
    SessionRevoked: "session-revoked",
    AccountDeleted: "account-deleted",
    EmailIdentifierChanged: "email-identifier-changed",
    MfaPhoneUnenrolled: "mfa-phone-unenrolled",
    AccountDeactivated: "account-deactivated",
};
/** The `mode` property determines the configuration method for enabling initiators. `custom` enables only the initiators listed in the selected_initiators array, `all` enables all current and future initiators. */
export const ClientOidcBackchannelLogoutInitiatorsModeEnum = {
    Custom: "custom",
    All: "all",
};
/** Method for discovering organizations during the `pre_login_prompt`. `email` allows users to find their organization by entering their email address and performing domain matching, while `organization_name` requires users to enter the organization name directly. These methods can be combined. */
export const ClientOrganizationDiscoveryEnum = {
    Email: "email",
    OrganizationName: "organization_name",
};
/** Defines how to proceed during an authentication transaction when `client.organization_usage: 'require'`. Can be `no_prompt` (default), `pre_login_prompt` or `post_login_prompt`. `post_login_prompt` requires `oidc_conformant: true`. */
export const ClientOrganizationRequireBehaviorEnum = {
    NoPrompt: "no_prompt",
    PreLoginPrompt: "pre_login_prompt",
    PostLoginPrompt: "post_login_prompt",
};
/** Defines how to proceed during an authentication transaction when `client.organization_usage: 'require'`. Can be `no_prompt` (default), `pre_login_prompt` or `post_login_prompt`. `post_login_prompt` requires `oidc_conformant: true`. */
export const ClientOrganizationRequireBehaviorPatchEnum = {
    NoPrompt: "no_prompt",
    PreLoginPrompt: "pre_login_prompt",
    PostLoginPrompt: "post_login_prompt",
};
/** Defines how to proceed during an authentication transaction with regards an organization. Can be `deny` (default), `allow` or `require`. */
export const ClientOrganizationUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** Defines how to proceed during an authentication transaction with regards an organization. Can be `deny` (default), `allow` or `require`. */
export const ClientOrganizationUsagePatchEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
export const ClientSessionTransferAllowedAuthenticationMethodsEnum = {
    Cookie: "cookie",
    Query: "query",
};
/** Indicates whether device binding security should be enforced for the app. If set to 'ip', the app will enforce device binding by IP, meaning that consumption of Session Transfer Token must be done from the same IP of the issuer. Likewise, if set to 'asn', device binding is enforced by ASN, meaning consumption of Session Transfer Token must be done from the same ASN as the issuer. If set to 'null', device binding is not enforced. Usually configured in the web application. */
export const ClientSessionTransferDeviceBindingEnum = {
    Ip: "ip",
    Asn: "asn",
    None: "none",
};
/** Defines the requested authentication method for the token endpoint. Can be `none` (public client without a client secret), `client_secret_post` (client uses HTTP POST parameters), or `client_secret_basic` (client uses HTTP Basic). */
export const ClientTokenEndpointAuthMethodEnum = {
    None: "none",
    ClientSecretPost: "client_secret_post",
    ClientSecretBasic: "client_secret_basic",
};
/** Defines the requested authentication method for the token endpoint. Can be `none` (public client without a client secret), `client_secret_post` (client uses HTTP POST parameters), or `client_secret_basic` (client uses HTTP Basic). */
export const ClientTokenEndpointAuthMethodOrNullEnum = {
    None: "none",
    ClientSecretPost: "client_secret_post",
    ClientSecretBasic: "client_secret_basic",
};
/** Order of precedence for attribute types */
export const ConnectionIdentifierPrecedenceEnum = {
    Email: "email",
    PhoneNumber: "phone_number",
    Username: "username",
};
/** The identity provider identifier for the connection */
export const ConnectionIdentityProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Aol: "aol",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Flickr: "flickr",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Miicard: "miicard",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    Renren: "renren",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    ThecitySandbox: "thecity-sandbox",
    Thecity: "thecity",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yammer: "yammer",
    Yandex: "yandex",
};
/** Signing key use, whether for encryption or signing */
export const ConnectionKeyUseEnum = {
    Encryption: "encryption",
    Signing: "signing",
};
/** Controls the UI used to challenge the user for their passkey. */
export const ConnectionPasskeyChallengeUiEnum = {
    Both: "both",
    Autofill: "autofill",
    Button: "button",
};
/** Password strength level */
export const ConnectionPasswordPolicyEnum = {
    None: "none",
    Low: "low",
    Fair: "fair",
    Good: "good",
    Excellent: "excellent",
};
/** When using an external IdP, this flag determines  whether 'name', 'given_name', 'family_name', 'nickname', and 'picture' attributes are updated. In addition, it also determines whether the user is created when user doesnt exist previously. Possible values are 'on_each_login' (default value, it configures the connection to automatically create the user if necessary and update the root attributes from the external IdP with each user login. When this setting is used, root attributes cannot be independently updated), 'on_first_login' (configures the connection to create the user and set the root attributes on first login only, allowing them to be independently updated thereafter), and 'never_on_login' (configures the connection not to create the user and not to set the root attributes from the external IdP, allowing them to be independently updated). */
export const ConnectionSetUserRootAttributesEnum = {
    OnEachLogin: "on_each_login",
    OnFirstLogin: "on_first_login",
    NeverOnLogin: "never_on_login",
};
export const ConnectionStrategyEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Aol: "aol",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Flickr: "flickr",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Miicard: "miicard",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    Renren: "renren",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    ThecitySandbox: "thecity-sandbox",
    Thecity: "thecity",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yammer: "yammer",
    Yandex: "yandex",
    Auth0Adldap: "auth0-adldap",
};
export const ConnectionUpstreamAliasEnum = {
    AcrValues: "acr_values",
    Audience: "audience",
    ClientId: "client_id",
    Display: "display",
    IdTokenHint: "id_token_hint",
    LoginHint: "login_hint",
    MaxAge: "max_age",
    Prompt: "prompt",
    Resource: "resource",
    ResponseMode: "response_mode",
    ResponseType: "response_type",
    UiLocales: "ui_locales",
};
/** Type of the encryption key to be created. */
export const CreateEncryptionKeyType = {
    CustomerProvidedRootKey: "customer-provided-root-key",
    TenantEncryptionKey: "tenant-encryption-key",
};
export const CreatedAuthenticationMethodTypeEnum = {
    Phone: "phone",
    Email: "email",
    Totp: "totp",
    WebauthnRoaming: "webauthn-roaming",
};
export const CreatedUserAuthenticationMethodTypeEnum = {
    Phone: "phone",
    Email: "email",
    Totp: "totp",
    WebauthnRoaming: "webauthn-roaming",
    Passkey: "passkey",
};
/** The HTTP header to fetch the client's IP address */
export const CustomDomainCustomClientIpHeaderEnum = {
    TrueClientIp: "true-client-ip",
    CfConnectingIp: "cf-connecting-ip",
    XForwardedFor: "x-forwarded-for",
    XAzureClientip: "x-azure-clientip",
    Empty: "",
};
/** Custom domain provisioning type. Must be `auth0_managed_certs` or `self_managed_certs`. */
export const CustomDomainProvisioningTypeEnum = {
    Auth0ManagedCerts: "auth0_managed_certs",
    SelfManagedCerts: "self_managed_certs",
};
/** Custom domain configuration status. Can be `failed`, `pending_verification`, or `ready`. */
export const CustomDomainStatusFilterEnum = {
    PendingVerification: "pending_verification",
    Ready: "ready",
    Failed: "failed",
};
/** Custom domain provisioning type. Can be `auth0_managed_certs` or `self_managed_certs`. */
export const CustomDomainTypeEnum = {
    Auth0ManagedCerts: "auth0_managed_certs",
    SelfManagedCerts: "self_managed_certs",
};
export const CustomProviderDeliveryMethodEnum = {
    Text: "text",
    Voice: "voice",
};
/** Key algorithm */
export const CustomSigningKeyAlgorithmEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
    Es256: "ES256",
    Es384: "ES384",
    Es512: "ES512",
    Ps256: "PS256",
    Ps384: "PS384",
    Ps512: "PS512",
};
/** Curve */
export const CustomSigningKeyCurveEnum = {
    P256: "P-256",
    P384: "P-384",
    P521: "P-521",
};
/** Key type */
export const CustomSigningKeyTypeEnum = {
    Ec: "EC",
    Rsa: "RSA",
};
/** Type of credentials to retrieve. Must be `public_key`, `refresh_token` or `rotating_refresh_token`. The property will default to `refresh_token` when paging is requested */
export const DeviceCredentialTypeEnum = {
    PublicKey: "public_key",
    RefreshToken: "refresh_token",
    RotatingRefreshToken: "rotating_refresh_token",
};
/** The Certificate Authority issued the certificate. */
export const DomainCertificateAuthorityEnum = {
    Letsencrypt: "letsencrypt",
    Googletrust: "googletrust",
};
/** The provisioning status of the certificate. */
export const DomainCertificateStatusEnum = {
    Provisioning: "provisioning",
    ProvisioningFailed: "provisioning_failed",
    Provisioned: "provisioned",
    RenewingFailed: "renewing_failed",
};
/** Domain verification method. */
export const DomainVerificationMethodNameEnum = {
    Cname: "cname",
    Txt: "txt",
};
/** The DNS record verification status. This field is relevant only for Custom Domains with Auth0-Managed Certificates. */
export const DomainVerificationStatusEnum = {
    Verified: "verified",
    Pending: "pending",
    Failed: "failed",
};
/** Name of the email provider. Can be `mailgun`, `mandrill`, `sendgrid`, `ses`, `sparkpost`, `smtp`, `azure_cs`, `ms365`, or `custom`. */
export const EmailProviderNameEnum = {
    Mailgun: "mailgun",
    Mandrill: "mandrill",
    Sendgrid: "sendgrid",
    Ses: "ses",
    Sparkpost: "sparkpost",
    Smtp: "smtp",
    AzureCs: "azure_cs",
    Ms365: "ms365",
    Custom: "custom",
};
/** Template name. Can be `verify_email`, `verify_email_by_code`, `reset_email`, `reset_email_by_code`, `welcome_email`, `blocked_account`, `stolen_credentials`, `enrollment_email`, `mfa_oob_code`, `user_invitation`, `async_approval`, `change_password` (legacy), or `password_reset` (legacy). */
export const EmailTemplateNameEnum = {
    VerifyEmail: "verify_email",
    VerifyEmailByCode: "verify_email_by_code",
    ResetEmail: "reset_email",
    ResetEmailByCode: "reset_email_by_code",
    WelcomeEmail: "welcome_email",
    BlockedAccount: "blocked_account",
    StolenCredentials: "stolen_credentials",
    EnrollmentEmail: "enrollment_email",
    MfaOobCode: "mfa_oob_code",
    UserInvitation: "user_invitation",
    ChangePassword: "change_password",
    PasswordReset: "password_reset",
    AsyncApproval: "async_approval",
};
/** Key state */
export const EncryptionKeyState = {
    PreActivation: "pre-activation",
    Active: "active",
    Deactivated: "deactivated",
    Destroyed: "destroyed",
};
/** Key type */
export const EncryptionKeyType = {
    CustomerProvidedRootKey: "customer-provided-root-key",
    EnvironmentRootKey: "environment-root-key",
    TenantMasterKey: "tenant-master-key",
    TenantEncryptionKey: "tenant-encryption-key",
};
/** Type of event */
export const EventStreamDeliveryEventTypeEnum = {
    UserCreated: "user.created",
    UserDeleted: "user.deleted",
    UserUpdated: "user.updated",
    OrganizationCreated: "organization.created",
    OrganizationUpdated: "organization.updated",
    OrganizationDeleted: "organization.deleted",
    OrganizationMemberAdded: "organization.member.added",
    OrganizationMemberDeleted: "organization.member.deleted",
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
    OrganizationConnectionAdded: "organization.connection.added",
    OrganizationConnectionUpdated: "organization.connection.updated",
    OrganizationConnectionRemoved: "organization.connection.removed",
    GroupCreated: "group.created",
    GroupUpdated: "group.updated",
    GroupDeleted: "group.deleted",
    GroupMemberAdded: "group.member.added",
    GroupMemberDeleted: "group.member.deleted",
};
/** AWS Region for EventBridge destination. */
export const EventStreamEventBridgeAwsRegionEnum = {
    AfSouth1: "af-south-1",
    ApEast1: "ap-east-1",
    ApEast2: "ap-east-2",
    ApNortheast1: "ap-northeast-1",
    ApNortheast2: "ap-northeast-2",
    ApNortheast3: "ap-northeast-3",
    ApSouth1: "ap-south-1",
    ApSouth2: "ap-south-2",
    ApSoutheast1: "ap-southeast-1",
    ApSoutheast2: "ap-southeast-2",
    ApSoutheast3: "ap-southeast-3",
    ApSoutheast4: "ap-southeast-4",
    ApSoutheast5: "ap-southeast-5",
    ApSoutheast6: "ap-southeast-6",
    ApSoutheast7: "ap-southeast-7",
    CaCentral1: "ca-central-1",
    CaWest1: "ca-west-1",
    EuCentral1: "eu-central-1",
    EuCentral2: "eu-central-2",
    EuNorth1: "eu-north-1",
    EuSouth1: "eu-south-1",
    EuSouth2: "eu-south-2",
    EuWest1: "eu-west-1",
    EuWest2: "eu-west-2",
    EuWest3: "eu-west-3",
    IlCentral1: "il-central-1",
    MeCentral1: "me-central-1",
    MeSouth1: "me-south-1",
    MxCentral1: "mx-central-1",
    SaEast1: "sa-east-1",
    UsGovEast1: "us-gov-east-1",
    UsGovWest1: "us-gov-west-1",
    UsEast1: "us-east-1",
    UsEast2: "us-east-2",
    UsWest1: "us-west-1",
    UsWest2: "us-west-2",
};
export const EventStreamEventTypeEnum = {
    UserCreated: "user.created",
    UserDeleted: "user.deleted",
    UserUpdated: "user.updated",
    OrganizationCreated: "organization.created",
    OrganizationUpdated: "organization.updated",
    OrganizationDeleted: "organization.deleted",
    OrganizationMemberAdded: "organization.member.added",
    OrganizationMemberDeleted: "organization.member.deleted",
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
    OrganizationConnectionAdded: "organization.connection.added",
    OrganizationConnectionUpdated: "organization.connection.updated",
    OrganizationConnectionRemoved: "organization.connection.removed",
    GroupCreated: "group.created",
    GroupUpdated: "group.updated",
    GroupDeleted: "group.deleted",
    GroupMemberAdded: "group.member.added",
    GroupMemberDeleted: "group.member.deleted",
};
/** Indicates whether the event stream is actively forwarding events. */
export const EventStreamStatusEnum = {
    Enabled: "enabled",
    Disabled: "disabled",
};
/** The type of event this test event represents. */
export const EventStreamTestEventTypeEnum = {
    UserCreated: "user.created",
    UserDeleted: "user.deleted",
    UserUpdated: "user.updated",
    OrganizationCreated: "organization.created",
    OrganizationUpdated: "organization.updated",
    OrganizationDeleted: "organization.deleted",
    OrganizationMemberAdded: "organization.member.added",
    OrganizationMemberDeleted: "organization.member.deleted",
    OrganizationMemberRoleAssigned: "organization.member.role.assigned",
    OrganizationMemberRoleDeleted: "organization.member.role.deleted",
    OrganizationConnectionAdded: "organization.connection.added",
    OrganizationConnectionUpdated: "organization.connection.updated",
    OrganizationConnectionRemoved: "organization.connection.removed",
    GroupCreated: "group.created",
    GroupUpdated: "group.updated",
    GroupDeleted: "group.deleted",
    GroupMemberAdded: "group.member.added",
    GroupMemberDeleted: "group.member.deleted",
};
export var FlowActionAuth0SendRequestParams;
(function (FlowActionAuth0SendRequestParams) {
    FlowActionAuth0SendRequestParams.Method = {
        Get: "GET",
        Post: "POST",
        Put: "PUT",
        Patch: "PATCH",
        Delete: "DELETE",
    };
})(FlowActionAuth0SendRequestParams || (FlowActionAuth0SendRequestParams = {}));
export var FlowActionFlowDelayFlowParams;
(function (FlowActionFlowDelayFlowParams) {
    FlowActionFlowDelayFlowParams.Units = {
        Seconds: "SECONDS",
        Minutes: "MINUTES",
        Hours: "HOURS",
        Days: "DAYS",
    };
})(FlowActionFlowDelayFlowParams || (FlowActionFlowDelayFlowParams = {}));
export var FlowActionHttpSendRequestParams;
(function (FlowActionHttpSendRequestParams) {
    FlowActionHttpSendRequestParams.Method = {
        Get: "GET",
        Post: "POST",
        Put: "PUT",
        Patch: "PATCH",
        Delete: "DELETE",
    };
    FlowActionHttpSendRequestParams.ContentType = {
        Json: "JSON",
        Form: "FORM",
        Xml: "XML",
    };
})(FlowActionHttpSendRequestParams || (FlowActionHttpSendRequestParams = {}));
export var FlowActionSalesforceSearchLeadsParams;
(function (FlowActionSalesforceSearchLeadsParams) {
    FlowActionSalesforceSearchLeadsParams.SearchField = {
        Email: "email",
        Name: "name",
        Phone: "phone",
        All: "all",
    };
})(FlowActionSalesforceSearchLeadsParams || (FlowActionSalesforceSearchLeadsParams = {}));
export var FlowActionSlackPostMessageParamsAttachment;
(function (FlowActionSlackPostMessageParamsAttachment) {
    FlowActionSlackPostMessageParamsAttachment.Color = {
        Good: "GOOD",
        Warning: "WARNING",
        Danger: "DANGER",
    };
})(FlowActionSlackPostMessageParamsAttachment || (FlowActionSlackPostMessageParamsAttachment = {}));
export var FlowActionWhatsappSendMessageParams;
(function (FlowActionWhatsappSendMessageParams) {
    FlowActionWhatsappSendMessageParams.Type = {
        Audio: "AUDIO",
        Contacts: "CONTACTS",
        Document: "DOCUMENT",
        Image: "IMAGE",
        Interactive: "INTERACTIVE",
        Location: "LOCATION",
        Sticker: "STICKER",
        Template: "TEMPLATE",
        Text: "TEXT",
    };
})(FlowActionWhatsappSendMessageParams || (FlowActionWhatsappSendMessageParams = {}));
export var FlowActionZapierTriggerWebhookParams;
(function (FlowActionZapierTriggerWebhookParams) {
    FlowActionZapierTriggerWebhookParams.Method = {
        Get: "GET",
        Post: "POST",
        Put: "PUT",
    };
})(FlowActionZapierTriggerWebhookParams || (FlowActionZapierTriggerWebhookParams = {}));
export const FlowsVaultConnectioSetupJwtAlgorithmEnum = {
    Hs256: "HS256",
    Hs384: "HS384",
    Hs512: "HS512",
    Rs256: "RS256",
    Rs384: "RS384",
    Rs512: "RS512",
    Es256: "ES256",
    Es384: "ES384",
    Es512: "ES512",
    Ps256: "PS256",
    Ps384: "PS384",
    Ps512: "PS512",
};
export const FormBlockImageConfigPositionEnum = {
    Left: "LEFT",
    Center: "CENTER",
    Right: "RIGHT",
};
export const FormBlockResendButtonConfigTextAlignmentEnum = {
    Left: "LEFT",
    Center: "CENTER",
    Right: "RIGHT",
};
export const FormFieldDateConfigFormatEnum = {
    Date: "DATE",
    Time: "TIME",
};
export const FormFieldFileConfigCategoryEnum = {
    Audio: "AUDIO",
    Video: "VIDEO",
    Image: "IMAGE",
    Document: "DOCUMENT",
    Archive: "ARCHIVE",
};
export const FormFieldFileConfigStorageTypeEnum = {
    Managed: "MANAGED",
    Custom: "CUSTOM",
};
export const FormFieldPasswordConfigHashEnum = {
    None: "NONE",
    Md5: "MD5",
    Sha1: "SHA1",
    Sha256: "SHA256",
    Sha512: "SHA512",
};
export const FormFieldPaymentConfigChargeOneOffCurrencyEnum = {
    Aud: "AUD",
    Cad: "CAD",
    Chf: "CHF",
    Eur: "EUR",
    Gbp: "GBP",
    Inr: "INR",
    Mxn: "MXN",
    Sek: "SEK",
    Usd: "USD",
};
export const FormsRequestParametersHydrateEnum = {
    FlowCount: "flow_count",
    Links: "links",
};
export var GetBruteForceSettingsResponseContent;
(function (GetBruteForceSettingsResponseContent) {
    let Shields;
    (function (Shields) {
        Shields.Item = {
            Block: "block",
            UserNotification: "user_notification",
        };
    })(Shields = GetBruteForceSettingsResponseContent.Shields || (GetBruteForceSettingsResponseContent.Shields = {}));
    /**
     * Account Lockout: Determines whether or not IP address is used when counting failed attempts.
     *           Possible values: <code>count_per_identifier_and_ip</code>, <code>count_per_identifier</code>.
     */
    GetBruteForceSettingsResponseContent.Mode = {
        CountPerIdentifierAndIp: "count_per_identifier_and_ip",
        CountPerIdentifier: "count_per_identifier",
    };
})(GetBruteForceSettingsResponseContent || (GetBruteForceSettingsResponseContent = {}));
export const GetFlowRequestParametersHydrateEnum = {
    FormCount: "form_count",
    Forms: "forms",
};
/** Optional. Specifies which factor the user must enroll with.<br />Note: Parameter can only be used with Universal Login; it cannot be used with Classic Login or custom MFA pages. */
export const GuardianEnrollmentFactorEnum = {
    PushNotification: "push-notification",
    Phone: "phone",
    Email: "email",
    Otp: "otp",
    WebauthnRoaming: "webauthn-roaming",
    WebauthnPlatform: "webauthn-platform",
};
/** Status of this enrollment. Can be `pending` or `confirmed`. */
export const GuardianEnrollmentStatus = {
    Pending: "pending",
    Confirmed: "confirmed",
};
/** Factor name. Can be `sms`, `push-notification`, `email`, `duo` `otp` `webauthn-roaming`, `webauthn-platform`, or `recovery-code`. */
export const GuardianFactorNameEnum = {
    PushNotification: "push-notification",
    Sms: "sms",
    Email: "email",
    Duo: "duo",
    Otp: "otp",
    WebauthnRoaming: "webauthn-roaming",
    WebauthnPlatform: "webauthn-platform",
    RecoveryCode: "recovery-code",
};
export const GuardianFactorPhoneFactorMessageTypeEnum = {
    Sms: "sms",
    Voice: "voice",
};
export const GuardianFactorsProviderPushNotificationProviderDataEnum = {
    Guardian: "guardian",
    Sns: "sns",
    Direct: "direct",
};
export const GuardianFactorsProviderSmsProviderEnum = {
    Auth0: "auth0",
    Twilio: "twilio",
    PhoneMessageHook: "phone-message-hook",
};
/** Retrieves hooks that match the trigger */
export const HookTriggerIdEnum = {
    CredentialsExchange: "credentials-exchange",
    PreUserRegistration: "pre-user-registration",
    PostUserRegistration: "post-user-registration",
    PostChangePassword: "post-change-password",
    SendPhoneMessage: "send-phone-message",
};
/** Identity provider name of the identity (e.g. `google-oauth2`). */
export const IdentityProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Aol: "aol",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Flickr: "flickr",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Miicard: "miicard",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    Renren: "renren",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    ThecitySandbox: "thecity-sandbox",
    Thecity: "thecity",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yammer: "yammer",
    Yandex: "yandex",
};
/** feature_type is the type of the integration. */
export const IntegrationFeatureTypeEnum = {
    Unspecified: "unspecified",
    Action: "action",
    SocialConnection: "social_connection",
    LogStream: "log_stream",
    SsoIntegration: "sso_integration",
    SmsProvider: "sms_provider",
};
export const IntegrationRequiredParamTypeEnum = {
    Unspecified: "UNSPECIFIED",
    String: "STRING",
};
/** Format of the file. Must be `json` or `csv`. */
export const JobFileFormatEnum = {
    Json: "json",
    Csv: "csv",
};
/** Datadog region */
export const LogStreamDatadogRegionEnum = {
    Us: "us",
    Eu: "eu",
    Us3: "us3",
    Us5: "us5",
};
/** The region in which the EventBridge event source will be created */
export const LogStreamEventBridgeSinkRegionEnum = {
    AfSouth1: "af-south-1",
    ApEast1: "ap-east-1",
    ApEast2: "ap-east-2",
    ApNortheast1: "ap-northeast-1",
    ApNortheast2: "ap-northeast-2",
    ApNortheast3: "ap-northeast-3",
    ApSouth1: "ap-south-1",
    ApSouth2: "ap-south-2",
    ApSoutheast1: "ap-southeast-1",
    ApSoutheast2: "ap-southeast-2",
    ApSoutheast3: "ap-southeast-3",
    ApSoutheast4: "ap-southeast-4",
    ApSoutheast5: "ap-southeast-5",
    ApSoutheast6: "ap-southeast-6",
    ApSoutheast7: "ap-southeast-7",
    CaCentral1: "ca-central-1",
    CaWest1: "ca-west-1",
    EuCentral1: "eu-central-1",
    EuCentral2: "eu-central-2",
    EuNorth1: "eu-north-1",
    EuSouth1: "eu-south-1",
    EuSouth2: "eu-south-2",
    EuWest1: "eu-west-1",
    EuWest2: "eu-west-2",
    EuWest3: "eu-west-3",
    IlCentral1: "il-central-1",
    MeCentral1: "me-central-1",
    MeSouth1: "me-south-1",
    MxCentral1: "mx-central-1",
    SaEast1: "sa-east-1",
    UsGovEast1: "us-gov-east-1",
    UsGovWest1: "us-gov-west-1",
    UsEast1: "us-east-1",
    UsEast2: "us-east-2",
    UsWest1: "us-west-1",
    UsWest2: "us-west-2",
};
/** Azure Region Name */
export const LogStreamEventGridRegionEnum = {
    Australiacentral: "australiacentral",
    Australiaeast: "australiaeast",
    Australiasoutheast: "australiasoutheast",
    Brazilsouth: "brazilsouth",
    Canadacentral: "canadacentral",
    Canadaeast: "canadaeast",
    Centralindia: "centralindia",
    Centralus: "centralus",
    Eastasia: "eastasia",
    Eastus: "eastus",
    Eastus2: "eastus2",
    Francecentral: "francecentral",
    Germanywestcentral: "germanywestcentral",
    Japaneast: "japaneast",
    Japanwest: "japanwest",
    Koreacentral: "koreacentral",
    Koreasouth: "koreasouth",
    Northcentralus: "northcentralus",
    Northeurope: "northeurope",
    Norwayeast: "norwayeast",
    Southafricanorth: "southafricanorth",
    Southcentralus: "southcentralus",
    Southeastasia: "southeastasia",
    Southindia: "southindia",
    Swedencentral: "swedencentral",
    Switzerlandnorth: "switzerlandnorth",
    Uaenorth: "uaenorth",
    Uksouth: "uksouth",
    Ukwest: "ukwest",
    Westcentralus: "westcentralus",
    Westeurope: "westeurope",
    Westindia: "westindia",
    Westus: "westus",
    Westus2: "westus2",
};
/** Category group name */
export const LogStreamFilterGroupNameEnum = {
    AuthLoginFail: "auth.login.fail",
    AuthLoginNotification: "auth.login.notification",
    AuthLoginSuccess: "auth.login.success",
    AuthLogoutFail: "auth.logout.fail",
    AuthLogoutSuccess: "auth.logout.success",
    AuthSignupFail: "auth.signup.fail",
    AuthSignupSuccess: "auth.signup.success",
    AuthSilentAuthFail: "auth.silent_auth.fail",
    AuthSilentAuthSuccess: "auth.silent_auth.success",
    AuthTokenExchangeFail: "auth.token_exchange.fail",
    AuthTokenExchangeSuccess: "auth.token_exchange.success",
    ManagementFail: "management.fail",
    ManagementSuccess: "management.success",
    ScimEvent: "scim.event",
    SystemNotification: "system.notification",
    UserFail: "user.fail",
    UserNotification: "user.notification",
    UserSuccess: "user.success",
    Actions: "actions",
    Other: "other",
};
/** HTTP JSON format */
export const LogStreamHttpContentFormatEnum = {
    Jsonarray: "JSONARRAY",
    Jsonlines: "JSONLINES",
    Jsonobject: "JSONOBJECT",
};
/** Mixpanel Region */
export const LogStreamMixpanelRegionEnum = {
    Us: "us",
    Eu: "eu",
};
export const LogStreamPiiLogFieldsEnum = {
    FirstName: "first_name",
    LastName: "last_name",
    Username: "username",
    Email: "email",
    Phone: "phone",
    Address: "address",
};
export const LogStreamPiiMethodEnum = {
    Mask: "mask",
    Hash: "hash",
};
/** The status of the log stream. Possible values: `active`, `paused`, `suspended` */
export const LogStreamStatusEnum = {
    Active: "active",
    Paused: "paused",
    Suspended: "suspended",
};
/** The MFA policy */
export const MfaPolicyEnum = {
    AllApplications: "all-applications",
    ConfidenceScore: "confidence-score",
};
/** Identifies the origin of the request as the Management API (management), Authentication API (authentication), or either (tenant) */
export const NetworkAclRuleScopeEnum = {
    Management: "management",
    Authentication: "authentication",
    Tenant: "tenant",
};
/** The verification status of the discovery domain. */
export const OrganizationDiscoveryDomainStatus = {
    Pending: "pending",
    Verified: "verified",
};
/** Defines whether organizations can be used with client credentials exchanges for this grant. */
export const OrganizationUsageEnum = {
    Deny: "deny",
    Allow: "allow",
    Require: "require",
};
/** Name of the prompt. */
export const PartialGroupsEnum = {
    Login: "login",
    LoginId: "login-id",
    LoginPassword: "login-password",
    LoginPasswordless: "login-passwordless",
    Signup: "signup",
    SignupId: "signup-id",
    SignupPassword: "signup-password",
    CustomizedConsent: "customized-consent",
};
/** The delivery method for the notification */
export const PhoneProviderDeliveryMethodEnum = {
    Text: "text",
    Voice: "voice",
};
/** Name of the phone notification provider */
export const PhoneProviderNameEnum = {
    Twilio: "twilio",
    Custom: "custom",
};
export const PhoneTemplateNotificationTypeEnum = {
    OtpVerify: "otp_verify",
    OtpEnroll: "otp_enroll",
    ChangePassword: "change_password",
    BlockedAccount: "blocked_account",
    PasswordBreach: "password_breach",
};
/** Applies to phone authentication methods only. The preferred communication method. */
export const PreferredAuthenticationMethodEnum = {
    Voice: "voice",
    Sms: "sms",
};
/** Name of the prompt */
export const PromptGroupNameEnum = {
    Login: "login",
    LoginId: "login-id",
    LoginPassword: "login-password",
    LoginPasswordless: "login-passwordless",
    LoginEmailVerification: "login-email-verification",
    Signup: "signup",
    SignupId: "signup-id",
    SignupPassword: "signup-password",
    PhoneIdentifierEnrollment: "phone-identifier-enrollment",
    PhoneIdentifierChallenge: "phone-identifier-challenge",
    EmailIdentifierChallenge: "email-identifier-challenge",
    ResetPassword: "reset-password",
    CustomForm: "custom-form",
    Consent: "consent",
    CustomizedConsent: "customized-consent",
    Logout: "logout",
    MfaPush: "mfa-push",
    MfaOtp: "mfa-otp",
    MfaVoice: "mfa-voice",
    MfaPhone: "mfa-phone",
    MfaWebauthn: "mfa-webauthn",
    MfaSms: "mfa-sms",
    MfaEmail: "mfa-email",
    MfaRecoveryCode: "mfa-recovery-code",
    Mfa: "mfa",
    Status: "status",
    DeviceFlow: "device-flow",
    EmailVerification: "email-verification",
    EmailOtpChallenge: "email-otp-challenge",
    Organizations: "organizations",
    Invitation: "invitation",
    Common: "common",
    Passkeys: "passkeys",
    Captcha: "captcha",
    BruteForceProtection: "brute-force-protection",
    AsyncApprovalFlow: "async-approval-flow",
};
/** Language to update. */
export const PromptLanguageEnum = {
    Am: "am",
    Ar: "ar",
    ArEg: "ar-EG",
    ArSa: "ar-SA",
    Az: "az",
    Bg: "bg",
    Bn: "bn",
    Bs: "bs",
    CaEs: "ca-ES",
    Cnr: "cnr",
    Cs: "cs",
    Cy: "cy",
    Da: "da",
    De: "de",
    El: "el",
    En: "en",
    EnCa: "en-CA",
    Es: "es",
    Es419: "es-419",
    EsAr: "es-AR",
    EsMx: "es-MX",
    Et: "et",
    EuEs: "eu-ES",
    Fa: "fa",
    Fi: "fi",
    Fr: "fr",
    FrCa: "fr-CA",
    FrFr: "fr-FR",
    GlEs: "gl-ES",
    Gu: "gu",
    He: "he",
    Hi: "hi",
    Hr: "hr",
    Hu: "hu",
    Hy: "hy",
    Id: "id",
    Is: "is",
    It: "it",
    Ja: "ja",
    Ka: "ka",
    Kk: "kk",
    Kn: "kn",
    Ko: "ko",
    Lt: "lt",
    Lv: "lv",
    Mk: "mk",
    Ml: "ml",
    Mn: "mn",
    Mr: "mr",
    Ms: "ms",
    My: "my",
    Nb: "nb",
    Nl: "nl",
    Nn: "nn",
    No: "no",
    Pa: "pa",
    Pl: "pl",
    Pt: "pt",
    PtBr: "pt-BR",
    PtPt: "pt-PT",
    Ro: "ro",
    Ru: "ru",
    Sk: "sk",
    Sl: "sl",
    So: "so",
    Sq: "sq",
    Sr: "sr",
    Sv: "sv",
    Sw: "sw",
    Ta: "ta",
    Te: "te",
    Th: "th",
    Tl: "tl",
    Tr: "tr",
    Uk: "uk",
    Ur: "ur",
    Vi: "vi",
    Zgh: "zgh",
    ZhCn: "zh-CN",
    ZhHk: "zh-HK",
    ZhTw: "zh-TW",
};
/** Algorithm which will be used with the credential. Can be one of RS256, RS384, PS256. If not specified, RS256 will be used. Applies to `public_key` credential type. */
export const PublicKeyCredentialAlgorithmEnum = {
    Rs256: "RS256",
    Rs384: "RS384",
    Ps256: "PS256",
};
/** Refresh token expiration types, one of: expiring, non-expiring */
export const RefreshTokenExpirationTypeEnum = {
    Expiring: "expiring",
    NonExpiring: "non-expiring",
};
/** Refresh token rotation types, one of: rotating, non-rotating */
export const RefreshTokenRotationTypeEnum = {
    Rotating: "rotating",
    NonRotating: "non-rotating",
};
/** Intended mechanism for Proof-of-Possession */
export const ResourceServerProofOfPossessionMechanismEnum = {
    Mtls: "mtls",
    Dpop: "dpop",
};
/** Defines the client flows policy for the resource server */
export const ResourceServerSubjectTypeAuthorizationClientPolicyEnum = {
    DenyAll: "deny_all",
    RequireClientGrant: "require_client_grant",
};
/** Defines the user flows policy for the resource server */
export const ResourceServerSubjectTypeAuthorizationUserPolicyEnum = {
    AllowAll: "allow_all",
    DenyAll: "deny_all",
    RequireClientGrant: "require_client_grant",
};
/** Dialect of access tokens that should be issued. `access_token` is a JWT containing standard Auth0 claims; `rfc9068_profile` is a JWT conforming to the IETF JWT Access Token Profile. `access_token_authz` and `rfc9068_profile_authz` additionally include RBAC permissions claims. */
export const ResourceServerTokenDialectResponseEnum = {
    AccessToken: "access_token",
    AccessTokenAuthz: "access_token_authz",
    Rfc9068Profile: "rfc9068_profile",
    Rfc9068ProfileAuthz: "rfc9068_profile_authz",
};
/** Dialect of issued access token. `access_token` is a JWT containing standard Auth0 claims; `rfc9068_profile` is a JWT conforming to the IETF JWT Access Token Profile. `access_token_authz` and `rfc9068_profile_authz` additionally include RBAC permissions claims. */
export const ResourceServerTokenDialectSchemaEnum = {
    AccessToken: "access_token",
    AccessTokenAuthz: "access_token_authz",
    Rfc9068Profile: "rfc9068_profile",
    Rfc9068ProfileAuthz: "rfc9068_profile_authz",
};
/** Algorithm used to encrypt the token. */
export const ResourceServerTokenEncryptionAlgorithmEnum = {
    RsaOaep256: "RSA-OAEP-256",
    RsaOaep384: "RSA-OAEP-384",
    RsaOaep512: "RSA-OAEP-512",
};
/** Selected Signing Algorithm */
export const RotateConnectionKeysSigningAlgEnum = {
    Rs256: "RS256",
    Rs512: "RS512",
    Ps256: "PS256",
    Es256: "ES256",
};
/** Name of the screen */
export const ScreenGroupNameEnum = {
    Login: "login",
    LoginId: "login-id",
    LoginPassword: "login-password",
    LoginPasswordlessEmailCode: "login-passwordless-email-code",
    LoginPasswordlessEmailLink: "login-passwordless-email-link",
    LoginPasswordlessSmsOtp: "login-passwordless-sms-otp",
    LoginEmailVerification: "login-email-verification",
    Signup: "signup",
    SignupId: "signup-id",
    SignupPassword: "signup-password",
    PhoneIdentifierEnrollment: "phone-identifier-enrollment",
    PhoneIdentifierChallenge: "phone-identifier-challenge",
    EmailIdentifierChallenge: "email-identifier-challenge",
    ResetPasswordRequest: "reset-password-request",
    ResetPasswordEmail: "reset-password-email",
    ResetPassword: "reset-password",
    ResetPasswordSuccess: "reset-password-success",
    ResetPasswordError: "reset-password-error",
    ResetPasswordMfaEmailChallenge: "reset-password-mfa-email-challenge",
    ResetPasswordMfaOtpChallenge: "reset-password-mfa-otp-challenge",
    ResetPasswordMfaPhoneChallenge: "reset-password-mfa-phone-challenge",
    ResetPasswordMfaPushChallengePush: "reset-password-mfa-push-challenge-push",
    ResetPasswordMfaRecoveryCodeChallenge: "reset-password-mfa-recovery-code-challenge",
    ResetPasswordMfaSmsChallenge: "reset-password-mfa-sms-challenge",
    ResetPasswordMfaVoiceChallenge: "reset-password-mfa-voice-challenge",
    ResetPasswordMfaWebauthnPlatformChallenge: "reset-password-mfa-webauthn-platform-challenge",
    ResetPasswordMfaWebauthnRoamingChallenge: "reset-password-mfa-webauthn-roaming-challenge",
    CustomForm: "custom-form",
    Consent: "consent",
    CustomizedConsent: "customized-consent",
    Logout: "logout",
    LogoutComplete: "logout-complete",
    LogoutAborted: "logout-aborted",
    MfaPushWelcome: "mfa-push-welcome",
    MfaPushEnrollmentQr: "mfa-push-enrollment-qr",
    MfaPushEnrollmentCode: "mfa-push-enrollment-code",
    MfaPushSuccess: "mfa-push-success",
    MfaPushChallengePush: "mfa-push-challenge-push",
    MfaPushList: "mfa-push-list",
    MfaOtpEnrollmentQr: "mfa-otp-enrollment-qr",
    MfaOtpEnrollmentCode: "mfa-otp-enrollment-code",
    MfaOtpChallenge: "mfa-otp-challenge",
    MfaVoiceEnrollment: "mfa-voice-enrollment",
    MfaVoiceChallenge: "mfa-voice-challenge",
    MfaPhoneChallenge: "mfa-phone-challenge",
    MfaPhoneEnrollment: "mfa-phone-enrollment",
    MfaWebauthnPlatformEnrollment: "mfa-webauthn-platform-enrollment",
    MfaWebauthnRoamingEnrollment: "mfa-webauthn-roaming-enrollment",
    MfaWebauthnPlatformChallenge: "mfa-webauthn-platform-challenge",
    MfaWebauthnRoamingChallenge: "mfa-webauthn-roaming-challenge",
    MfaWebauthnChangeKeyNickname: "mfa-webauthn-change-key-nickname",
    MfaWebauthnEnrollmentSuccess: "mfa-webauthn-enrollment-success",
    MfaWebauthnError: "mfa-webauthn-error",
    MfaWebauthnNotAvailableError: "mfa-webauthn-not-available-error",
    MfaCountryCodes: "mfa-country-codes",
    MfaSmsEnrollment: "mfa-sms-enrollment",
    MfaSmsChallenge: "mfa-sms-challenge",
    MfaSmsList: "mfa-sms-list",
    MfaEmailChallenge: "mfa-email-challenge",
    MfaEmailList: "mfa-email-list",
    MfaRecoveryCodeEnrollment: "mfa-recovery-code-enrollment",
    MfaRecoveryCodeChallengeNewCode: "mfa-recovery-code-challenge-new-code",
    MfaRecoveryCodeChallenge: "mfa-recovery-code-challenge",
    MfaDetectBrowserCapabilities: "mfa-detect-browser-capabilities",
    MfaEnrollResult: "mfa-enroll-result",
    MfaLoginOptions: "mfa-login-options",
    MfaBeginEnrollOptions: "mfa-begin-enroll-options",
    Status: "status",
    DeviceCodeActivation: "device-code-activation",
    DeviceCodeActivationAllowed: "device-code-activation-allowed",
    DeviceCodeActivationDenied: "device-code-activation-denied",
    DeviceCodeConfirmation: "device-code-confirmation",
    EmailVerificationResult: "email-verification-result",
    EmailOtpChallenge: "email-otp-challenge",
    OrganizationSelection: "organization-selection",
    OrganizationPicker: "organization-picker",
    PreLoginOrganizationPicker: "pre-login-organization-picker",
    AcceptInvitation: "accept-invitation",
    RedeemTicket: "redeem-ticket",
    PasskeyEnrollment: "passkey-enrollment",
    PasskeyEnrollmentLocal: "passkey-enrollment-local",
    InterstitialCaptcha: "interstitial-captcha",
    BruteForceProtectionUnblock: "brute-force-protection-unblock",
    BruteForceProtectionUnblockSuccess: "brute-force-protection-unblock-success",
    BruteForceProtectionUnblockFailure: "brute-force-protection-unblock-failure",
    AsyncApprovalError: "async-approval-error",
    AsyncApprovalWrongUser: "async-approval-wrong-user",
    AsyncApprovalAccepted: "async-approval-accepted",
    AsyncApprovalDenied: "async-approval-denied",
};
/** The version of the search engine */
export const SearchEngineVersionsEnum = {
    V1: "v1",
    V2: "v2",
    V3: "v3",
};
export const SelfServiceProfileAllowedStrategyEnum = {
    Oidc: "oidc",
    Samlp: "samlp",
    Waad: "waad",
    GoogleApps: "google-apps",
    Adfs: "adfs",
    Okta: "okta",
    KeycloakSamlp: "keycloak-samlp",
    Pingfederate: "pingfederate",
};
/** Whether the end user should complete the domain verification step. Possible values are 'none' (the step is not shown to the user), 'optional' (the user may add a domain alias in the domain verification step) or 'required' (the user must add a domain alias in order to enable the connection). Defaults to 'none'. */
export const SelfServiceProfileSsoTicketDomainVerificationEnum = {
    None: "none",
    Optional: "optional",
    Required: "required",
};
/** The protocol used to connect to the the default application */
export const SelfServiceProfileSsoTicketIdpInitiatedClientProtocolEnum = {
    Samlp: "samlp",
    Wsfed: "wsfed",
    Oauth2: "oauth2",
};
export const SelfServiceProfileSsoTicketProvisioningScopeEnum = {
    GetUsers: "get:users",
    PostUsers: "post:users",
    PutUsers: "put:users",
    PatchUsers: "patch:users",
    DeleteUsers: "delete:users",
};
/** [Private Early Access] The persistence mode of the session cookie. When set to "non-persistent" (ephemeral), the cookie will be deleted when the browser is closed. When set to "persistent", the cookie will be stored until it expires or is deleted by the user. */
export const SessionCookieMetadataModeEnum = {
    NonPersistent: "non-persistent",
    Persistent: "persistent",
};
/** Behavior of the session cookie */
export const SessionCookieModeEnum = {
    Persistent: "persistent",
    NonPersistent: "non-persistent",
};
/** Algorithm used to sign JWTs. Can be `HS256` or `RS256`. `PS256` available via addon. */
export const SigningAlgorithmEnum = {
    Hs256: "HS256",
    Rs256: "RS256",
    Rs512: "RS512",
    Ps256: "PS256",
};
export const SignupStatusEnum = {
    Required: "required",
    Optional: "optional",
    Inactive: "inactive",
};
export const SupportedLocales = {
    Am: "am",
    Ar: "ar",
    ArEg: "ar-EG",
    ArSa: "ar-SA",
    Az: "az",
    Bg: "bg",
    Bn: "bn",
    Bs: "bs",
    CaEs: "ca-ES",
    Cnr: "cnr",
    Cs: "cs",
    Cy: "cy",
    Da: "da",
    De: "de",
    El: "el",
    En: "en",
    EnCa: "en-CA",
    Es: "es",
    Es419: "es-419",
    EsAr: "es-AR",
    EsMx: "es-MX",
    Et: "et",
    EuEs: "eu-ES",
    Fa: "fa",
    Fi: "fi",
    Fr: "fr",
    FrCa: "fr-CA",
    FrFr: "fr-FR",
    GlEs: "gl-ES",
    Gu: "gu",
    He: "he",
    Hi: "hi",
    Hr: "hr",
    Hu: "hu",
    Hy: "hy",
    Id: "id",
    Is: "is",
    It: "it",
    Ja: "ja",
    Ka: "ka",
    Kk: "kk",
    Kn: "kn",
    Ko: "ko",
    Lt: "lt",
    Lv: "lv",
    Mk: "mk",
    Ml: "ml",
    Mn: "mn",
    Mr: "mr",
    Ms: "ms",
    My: "my",
    Nb: "nb",
    Nl: "nl",
    Nn: "nn",
    No: "no",
    Pa: "pa",
    Pl: "pl",
    Pt: "pt",
    PtBr: "pt-BR",
    PtPt: "pt-PT",
    Ro: "ro",
    Ru: "ru",
    Sk: "sk",
    Sl: "sl",
    So: "so",
    Sq: "sq",
    Sr: "sr",
    Sv: "sv",
    Sw: "sw",
    Ta: "ta",
    Te: "te",
    Th: "th",
    Tl: "tl",
    Tr: "tr",
    Uk: "uk",
    Ur: "ur",
    Vi: "vi",
    Zgh: "zgh",
    ZhCn: "zh-CN",
    ZhHk: "zh-HK",
    ZhTw: "zh-TW",
};
export const SuspiciousIpThrottlingShieldsEnum = {
    Block: "block",
    AdminNotification: "admin_notification",
};
/** Character set used to generate a User Code. Can be `base20` or `digits`. */
export const TenantSettingsDeviceFlowCharset = {
    Base20: "base20",
    Digits: "digits",
};
export const TwilioProviderDeliveryMethodEnum = {
    Text: "text",
    Voice: "voice",
};
/** Which login experience to use. Can be `new` or `classic`. */
export const UniversalLoginExperienceEnum = {
    New: "new",
    Classic: "classic",
};
export var UpdateBruteForceSettingsResponseContent;
(function (UpdateBruteForceSettingsResponseContent) {
    let Shields;
    (function (Shields) {
        Shields.Item = {
            Block: "block",
            UserNotification: "user_notification",
        };
    })(Shields = UpdateBruteForceSettingsResponseContent.Shields || (UpdateBruteForceSettingsResponseContent.Shields = {}));
    /**
     * Account Lockout: Determines whether or not IP address is used when counting failed attempts.
     *           Possible values: <code>count_per_identifier_and_ip</code>, <code>count_per_identifier</code>.
     */
    UpdateBruteForceSettingsResponseContent.Mode = {
        CountPerIdentifierAndIp: "count_per_identifier_and_ip",
        CountPerIdentifier: "count_per_identifier",
    };
})(UpdateBruteForceSettingsResponseContent || (UpdateBruteForceSettingsResponseContent = {}));
/** OIDC mapping override for this strategy */
export const UserAttributeProfileUserIdOidcStrategyOverrideMapping = {
    Sub: "sub",
    Oid: "oid",
    Email: "email",
};
export const UserAuthenticationMethodPropertiesEnum = {
    Totp: "totp",
    Push: "push",
    Sms: "sms",
    Voice: "voice",
};
/** Authentication method for this enrollment. Can be `authenticator`, `guardian`, `sms`, `webauthn-roaming`, or `webauthn-platform`. */
export const UserEnrollmentAuthMethodEnum = {
    Authenticator: "authenticator",
    Guardian: "guardian",
    Sms: "sms",
    WebauthnPlatform: "webauthn-platform",
    WebauthnRoaming: "webauthn-roaming",
};
/** Status of this enrollment. Can be `pending` or `confirmed`. */
export const UserEnrollmentStatusEnum = {
    Pending: "pending",
    Confirmed: "confirmed",
};
/** The type of identity provider */
export const UserIdentityProviderEnum = {
    Ad: "ad",
    Adfs: "adfs",
    Amazon: "amazon",
    Apple: "apple",
    Dropbox: "dropbox",
    Bitbucket: "bitbucket",
    Aol: "aol",
    Auth0Oidc: "auth0-oidc",
    Auth0: "auth0",
    Baidu: "baidu",
    Bitly: "bitly",
    Box: "box",
    Custom: "custom",
    Daccount: "daccount",
    Dwolla: "dwolla",
    Email: "email",
    EvernoteSandbox: "evernote-sandbox",
    Evernote: "evernote",
    Exact: "exact",
    Facebook: "facebook",
    Fitbit: "fitbit",
    Flickr: "flickr",
    Github: "github",
    GoogleApps: "google-apps",
    GoogleOauth2: "google-oauth2",
    Instagram: "instagram",
    Ip: "ip",
    Line: "line",
    Linkedin: "linkedin",
    Miicard: "miicard",
    Oauth1: "oauth1",
    Oauth2: "oauth2",
    Office365: "office365",
    Oidc: "oidc",
    Okta: "okta",
    Paypal: "paypal",
    PaypalSandbox: "paypal-sandbox",
    Pingfederate: "pingfederate",
    Planningcenter: "planningcenter",
    Renren: "renren",
    SalesforceCommunity: "salesforce-community",
    SalesforceSandbox: "salesforce-sandbox",
    Salesforce: "salesforce",
    Samlp: "samlp",
    Sharepoint: "sharepoint",
    Shopify: "shopify",
    Shop: "shop",
    Sms: "sms",
    Soundcloud: "soundcloud",
    ThecitySandbox: "thecity-sandbox",
    Thecity: "thecity",
    Thirtysevensignals: "thirtysevensignals",
    Twitter: "twitter",
    Untappd: "untappd",
    Vkontakte: "vkontakte",
    Waad: "waad",
    Weibo: "weibo",
    Windowslive: "windowslive",
    Wordpress: "wordpress",
    Yahoo: "yahoo",
    Yammer: "yammer",
    Yandex: "yandex",
};
/** The multi-factor provider. Supported values 'duo' or 'google-authenticator' */
export const UserMultifactorProviderEnum = {
    Duo: "duo",
    GoogleAuthenticator: "google-authenticator",
};
export const VerificationMethodEnum = {
    Link: "link",
    Otp: "otp",
};
